File: BBIS.Api/ClientApp/src/models/TestOrder/TestOrderDto.ts
import { BloodScreeningTestOrderDto } from "./BloodScreeningTestOrderDto";
import { BloodTypingTestOrderDto } from "./BloodTypingTestOrderDto";
import { CoombsTestOrderDto } from "./CoombsTestOrderDto";
import { CrossMatchTestOrderDto } from "./CrossMatchTestOrderDto";
import { GroupCrossMatchTestOrderViewDto } from "./GroupCrossMatchTestOrderViewDto";

export interface TestOrderDto {
    TestOrderId: Guid;
    PatientId: Guid | null;
    PatientName: string;
    ReservationId: Guid | null;
    TestCompleted: boolean;
    Remarks: string;
    PerformedBy: Guid | null;
    ReviewedBy: Guid | null;
    ValidatedBy: Guid | null;
    PathologistId: Guid | null;
    CoombsTestOrder: CoombsTestOrderDto | null;
    BloodScreeningTestOrder: BloodScreeningTestOrderDto | null;
    BloodTypingTestOrder: BloodTypingTestOrderDto | null;
    GroupCrossMatchTestOrders: Array<GroupCrossMatchTestOrderViewDto>;
}

File: BBIS.Application/DTOs/TestOrder/TestOrderDto.cs
namespace BBIS.Application.DTOs.TestOrder
{
    public class TestOrderDto
    {
        public Guid TestOrderId { get; set; }
        public Guid PatientId { get; set; }
        public string PatientName { get; set; }
        public Guid ReservationId { get; set; }
        public bool TestCompleted { get; set; }
        public string TestOrderNumber { get; set; }
        public string Remarks { get; set; }

        public Guid? PerformedBy { get; set; }
        public Guid? ReviewedBy { get; set; }
        public Guid? ValidatedBy { get; set; }
        public Guid? PathologistId { get; set; }

        public CoombsTestOrderDto CoombsTestOrder { get; set; }
        public BloodScreeningTestOrderDto BloodScreeningTestOrder { get; set; }
        public BloodTypingTestOrderDto BloodTypingTestOrder { get; set; }
        public List<GroupCrossMatchTestOrderViewDto> GroupCrossMatchTestOrders { get; set; }
    }
}
